### PR TITLE
SYN-15 Create SDK methods for woocommerce

### PR DESCRIPTION
**Pull request by dhruv301**
Created New WooCommerce project for SDK methods and created POJO's for webhook,update inventory and authentication. Created one file of constantcode and class for deserialization.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <maven.compiler.source>11</maven.compiler.source>
+        <maven.compiler.target>11</maven.compiler.target>
+        <version.plugin.maven-release-plugin>2.5.2</version.plugin.maven-release-plugin>
+        <version.plugin.nexus-staging-maven-plugin>1.6.5</version.plugin.nexus-staging-maven-plugin>
+        <version.plugin.maven-gpg-plugin>1.6</version.plugin.maven-gpg-plugin>
+        <source.encoding>UTF-8</source.encoding>
+    </properties>
+
+    <dependencies>
```
Comment (manan-dft) [inline pom.xml:47]: Why do you need spring boot starter?

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <version>1.18.24</version>
+        </dependency>
+
```
Comment (manan-dft) [inline pom.xml]: remove the space

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-databind</artifactId>
+            <version>2.13.4</version>
+        </dependency>
+
```
Comment (manan-dft) [inline pom.xml]: remove the space

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpclient</artifactId>
+            <version>4.5.13</version>
+        </dependency>
+        <dependency>
+            <groupId>javax.xml.bind</groupId>
```
Comment (manan-dft) [inline pom.xml]: what is use of this dependency ? it is needed ?

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-parent</artifactId>
+        <version>2.7.7</version>
+        <relativePath/>
+    </parent>
+
+
+    <groupId>io.github.dftrakesh</groupId>
+    <artifactId>woocommerce</artifactId>
+    <version>1.0-SNAPSHOT</version>
```
Comment (manan-dft) [inline pom.xml]: version should be 1.0

File: pom.xml (hunk 1 above)
Comment (dhruv301) [reply pom.xml:47]: I Started Work on next Ticket before Submitting PR. In That I need @Controller So for that i added that

File: pom.xml (hunk 2 above)
Comment (dhruv301) [reply pom.xml]: Ok

File: pom.xml (hunk 3 above)
Comment (dhruv301) [reply pom.xml]: Ok

File: pom.xml (hunk 4 above)
Comment (dhruv301) [reply pom.xml]: Which one? Javax.xml.bind?

File: pom.xml (hunk 5 above)
Comment (dhruv301) [reply pom.xml]: Ok will change it

File: pom.xml (hunk 1 above)
Comment (manan-dft) [reply pom.xml:47]: The next ticket is for working on another module. the controller will never come in sdk.

File: pom.xml (hunk 4 above)
Comment (manan-dft) [reply pom.xml]: You can see that on which line I have posted comment.

File: pom.xml (hunk 1 above)
Comment (dhruv301) [reply pom.xml:47]: Ok i will remove it

File: pom.xml (hunk 4 above)
Comment (dhruv301) [reply pom.xml]: No it's not needed. i removed that

File: src/main/java/io/github/dft/woocommerce/WooCommerceSdk.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        return objectMapper.writeValueAsString(body);
+    }
+
+    @SneakyThrows
+    protected <T> T getRequestWrapped(HttpRequest request, Class<T> tClass) {
+        return client.sendAsync(request, HttpResponse.BodyHandlers.ofString())
+                .thenComposeAsync(response -> tryResend(client, request, HttpResponse.BodyHandlers.ofString(), response,
+                        1))
+                .thenApplyAsync(stringHttpResponse -> {
+                    System.out.println("body: " + stringHttpResponse.body());
```
Comment (manan-dft) [inline src/main/java/io/github/dft/woocommerce/WooCommerceSdk.java]: remote sout

File: src/main/java/io/github/dft/woocommerce/model/authenticationapi/AccessCredential.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class AccessCredential {
+
+    private String shopDomain;
+    private String keyId;
+    private String userId;
+    private String consumerKey;
+    private String consumerSecret;
+    private String keyPermissions;
+
```
Comment (manan-dft) [inline src/main/java/io/github/dft/woocommerce/model/authenticationapi/AccessCredential.java]: remove line

File: src/main/java/io/github/dft/woocommerce/model/authenticationapi/AccessCredential.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder(toBuilder = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class AccessCredential {
+
```
Comment (manan-dft) [inline src/main/java/io/github/dft/woocommerce/model/authenticationapi/AccessCredential.java]: remove line

File: src/main/java/io/github/dft/woocommerce/model/common/DateDeserializer.java [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+public class DateDeserializer extends JsonDeserializer<LocalDateTime> {
+
+    @Override
+    public LocalDateTime deserialize(JsonParser jp, DeserializationContext ctxt) {
+        try {
+            String timestamp = jp.getText().trim();
+            DateTimeFormatter isoFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss");
+            return LocalDateTime.parse(timestamp, isoFormatter);
+        }catch (Exception exception){
```
Comment (manan-dft) [inline src/main/java/io/github/dft/woocommerce/model/common/DateDeserializer.java]: format the code

File: src/main/java/io/github/dft/woocommerce/WooCommerceSdk.java (hunk 6 above)
Comment (dhruv301) [reply src/main/java/io/github/dft/woocommerce/WooCommerceSdk.java]: Ok

File: src/main/java/io/github/dft/woocommerce/model/authenticationapi/AccessCredential.java (hunk 8 above)
Comment (dhruv301) [reply src/main/java/io/github/dft/woocommerce/model/authenticationapi/AccessCredential.java]: Ok

File: src/main/java/io/github/dft/woocommerce/model/authenticationapi/AccessCredential.java (hunk 7 above)
Comment (dhruv301) [reply src/main/java/io/github/dft/woocommerce/model/authenticationapi/AccessCredential.java]: Ok

**Review by manan-dft — APPROVED** [review]

**Pull request merged**